### PR TITLE
Netdata/packaging: Add documentation for binary packages, plus draft table for distributions support

### PR DESCRIPTION
**Pull request by paulkatsoulakis**
##### Summary
Here's the first version of the packaging documentation.

Note: also added one small nit on the package dependencies that require python and we only mention it on one distribution set

##### Component Name
netdata/packaging
Documentation

### CODE REVIEW

File: packaging/installer/README.md [hunk 1]
```diff
@@ -293,6 +293,28 @@ To apply the changes you made, you have to restart Netdata.
 
 ---
 
+### Binary Packages
+![](https://raw.githubusercontent.com/netdata/netdata/master/web/gui/images/packaging-beta-tag.svg?sanitize=true)
+
+Netdata is proud to present it's own flavour of binary packages for the most common operating systems that comply with .RPM and .DEB packaging formats.
```
Comment (cakrit) [inline packaging/installer/README.md]: No need for the `Netdata is proud to present `

Comment (paulkatsoulakis) [reply packaging/installer/README.md]: done

File: packaging/DISTRIBUTIONS.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+that we currently supporting and we are working on documenting our current state so that our users
+have complete visibility over the range of support
+
+Distribution | Family| Code health | Installer support | Kickstart support | Binary packaging support | Integrity testing (CI) | Functionality testing (CI) | Community support
+:------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :--------------------
+14.04.6 LTS (Trusty Tahr) | Ubuntu |  |  |  |  |  |  | 
+16.04.6 LTS (Xenial Xerus) | Ubuntu |  |  |  |  |  |  | 
+18.04.2 LTS (Bionic Beaver) | Ubuntu |  |  |  |  |  |  | 
+19.04 (Disco Dingo) Latest | Ubuntu |  |  |  |  |  |  | 
+Debian 7 (Wheezy) | Debian |  |  |  |  |  |  |
```
Comment (ilyam8) [inline packaging/DISTRIBUTIONS.md]: debian 7 is rip

File: packaging/DISTRIBUTIONS.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Distribution | Family| Code health | Installer support | Kickstart support | Binary packaging support | Integrity testing (CI) | Functionality testing (CI) | Community support
+:------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :------------------: | :--------------------
+14.04.6 LTS (Trusty Tahr) | Ubuntu |  |  |  |  |  |  | 
+16.04.6 LTS (Xenial Xerus) | Ubuntu |  |  |  |  |  |  | 
+18.04.2 LTS (Bionic Beaver) | Ubuntu |  |  |  |  |  |  | 
+19.04 (Disco Dingo) Latest | Ubuntu |  |  |  |  |  |  | 
+Debian 7 (Wheezy) | Debian |  |  |  |  |  |  | 
+Debian 8 (Jessie) | Debian |  |  |  |  |  |  | 
+Debian 9 (Stretch) | Debian |  |  |  |  |  |  | 
+Debian 10 (Buster) **Not released | Debian |  |  |  |  |  |  |
```
Comment (ilyam8) [inline packaging/DISTRIBUTIONS.md]: debin 10 released :tada:  - [proof link](https://www.debian.org/News/2019/20190706)

File: packaging/DISTRIBUTIONS.md (hunk 2 above)
Comment (paulkatsoulakis) [reply packaging/DISTRIBUTIONS.md]: Sure, but [this](https://wiki.debian.org/LTS/Extended) mentions extended support starting with 7

File: packaging/DISTRIBUTIONS.md (hunk 3 above)
Comment (paulkatsoulakis) [reply packaging/DISTRIBUTIONS.md]: Good point, updated

File: packaging/DISTRIBUTIONS.md (hunk 2 above)
Comment (paulkatsoulakis) [reply packaging/DISTRIBUTIONS.md]: In general the idea is that we follow LTS from the official distributions, but we ought to support our users that still remain on versions that recently got outdated anyway.

Comment (ilyam8) [reply packaging/DISTRIBUTIONS.md]: This ELTS thing for debian 7 is not official tho. It is commercial service

> This service provides security support for Debian 7 Wheezy, albeit only on the subset of packages used by the customers of this service.

**Comment by thiagoftsm** [comment]
I agree with @ilyam8 , according to the Debian the support finished ( https://www.debian.org/releases/wheezy/), so I believe we can remove it of the table.

File: packaging/DISTRIBUTIONS.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+CentOS 7.* | Cent OS | |  |  |  |  |  |  | 
+CentOS 8.* | Cent OS | |  |  |  |  |  |  | 
+Open SuSE Leap 15.0 | Open SuSE |  |  |  |  |  |  | 
+Open SuSE Leap 15.1 | Open SuSE |  |  |  |  |  |  | 
+Open SuSE Tumbleweed (latest) | Open SuSE |  |  |  |  |  |  | 
+SuSE Enterprise Linux Server 11 | SLES |  |  |  |  |  |  | 
+SuSE Enterprise Linux Server 12 | SLES |  |  |  |  |  |  | 
+SuSE Enterprise Linux Server 15 | SLES |  |  |  |  |  |  | 
+Arch Linux (latest) | Arch |  |  |  |  |  |  | 
+All other linux | Other | |  |  |  |  |  |
```
Comment (ilyam8) [inline packaging/DISTRIBUTIONS.md]: Do we need `all other linux`? Sounds to vague. How can you define `Code health` or `Functionality testing (CI)` for `all other linux`? Is debian6 in all other linux? :smile:

Comment (paulkatsoulakis) [reply packaging/DISTRIBUTIONS.md]: We have the static binary, that is for all other linux. Which reminds me i need to add a column about architecture

Comment (thiagoftsm) [reply packaging/DISTRIBUTIONS.md]: I think we can remove the "all other linux" of this table, and we can write a paragraph saying that people can compile netdata on "All linux" case they have available the libraries.

**Comment by paulkatsoulakis** [comment]
> I agree with @ilyam8 , according to the Debian the support finished ( https://www.debian.org/releases/wheezy/), so I believe we can remove it of the table.

Check my reply to ilya @thiagoftsm

File: packaging/DISTRIBUTIONS.md (hunk 4 above)
Comment (paulkatsoulakis) [reply packaging/DISTRIBUTIONS.md]: The various IoT Devices we want to support also probably fall on "all other linux", until we fine grain and go ahead and do more targeted support over those too

Comment (ilyam8) [reply packaging/DISTRIBUTIONS.md]: > In the following table we provide the list of officially supported operating systems.

So we are officially supporting all distributions? Because specific + other = all.

I tend to agree with @thiagoftsm, maybe it would be better to move `other` outside the table. Not sure tho.

File: packaging/DISTRIBUTIONS.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# Netdata distribution support matrix
```
Comment (cakrit) [inline packaging/DISTRIBUTIONS.md]: Since you added a new file, you'll need to update `docs/generator/buildyaml.sh` as well. Otherwise, the link won't work.
See https://deploy-preview-6422--netdata.netlify.com/DISTRIBUTIONS/, it's a 404.

Comment (paulkatsoulakis) [reply packaging/DISTRIBUTIONS.md]: wow, i didn't know about that, checking

File: packaging/DISTRIBUTIONS.md (hunk 4 above)
Comment (paulkatsoulakis) [reply packaging/DISTRIBUTIONS.md]: "All other linux" implies the static 64 binary for any other linux distribution. I want to depict for that case what things we cover exactly.

Don't forget this is just a support matrix, so that we can quickly tell what we do for each distro.
Installation instructions explaining how to install netdata for all cases

File: packaging/DISTRIBUTIONS.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+![](https://raw.githubusercontent.com/netdata/netdata/master/web/gui/images/packaging-beta-tag.svg?sanitize=true)
+
+In the following table we provide the list of officially supported operating systems.
```
Comment (joelhans) [inline packaging/DISTRIBUTIONS.md]: Maybe replace this paragraph with this: `In the following table we've listed Netdata's official supported operating systems. We detail the distributions, flavors, and the level of support Netdata is currently capable to provide.`

File: packaging/DISTRIBUTIONS.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+We detail the Distributions, flavours and the level of support Netdata is currently capable to provide.
+
+Kindly note that the following table is a work in progress. We have concluded on the list of distributions
+that we currently supporting and we are working on documenting our current state so that our users
+have complete visibility over the range of support
```
Comment (joelhans) [inline packaging/DISTRIBUTIONS.md]: Missing period at the end here.

File: packaging/DISTRIBUTIONS.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
+# Netdata distribution support matrix
+![](https://raw.githubusercontent.com/netdata/netdata/master/web/gui/images/packaging-beta-tag.svg?sanitize=true)
+
+In the following table we provide the list of officially supported operating systems.
+We detail the Distributions, flavours and the level of support Netdata is currently capable to provide.
+
+Kindly note that the following table is a work in progress. We have concluded on the list of distributions
```
Comment (joelhans) [inline packaging/DISTRIBUTIONS.md]: No need for "Kindly note that". Just start with "The following table..."

File: packaging/installer/README.md [hunk 9]
```diff
@@ -293,6 +293,28 @@ To apply the changes you made, you have to restart Netdata.
...
+![](https://raw.githubusercontent.com/netdata/netdata/master/web/gui/images/packaging-beta-tag.svg?sanitize=true)
+
+We provide our own flavour of binary packages for the most common operating systems that comply with .RPM and .DEB packaging formats.
+We have currently released .RPM versions with version [1.16.0](https://github.com/netdata/netdata/releases/tag/v1.16.0). We are planning to release packages following the .DEB format on one of the following releases. Our current packaging infrastructure provider is [Package Cloud](https://packagecloud.io).
+Netdata is committed to support installation of our solution to all operating systems.
+This is a constant battle for Netdata, as we strive to automate and make things easier for our users.
+For the operating system support matrix, please visit our [distributions](../DISTRIBUTIONS.md) support page.
+
+We provide two separate repositories, one for our stable releases and one for our nightly releases.
+1. Stable releases
```
Comment (joelhans) [inline packaging/installer/README.md]: You need another return before the ordered list. On the preview it's combined with the sentence just above.

File: packaging/installer/README.md [hunk 10]
```diff
@@ -293,6 +293,28 @@ To apply the changes you made, you have to restart Netdata.
 
 ---
 
+### Binary Packages
+![](https://raw.githubusercontent.com/netdata/netdata/master/web/gui/images/packaging-beta-tag.svg?sanitize=true)
+
+We provide our own flavour of binary packages for the most common operating systems that comply with .RPM and .DEB packaging formats.
+We have currently released .RPM versions with version [1.16.0](https://github.com/netdata/netdata/releases/tag/v1.16.0). We are planning to release packages following the .DEB format on one of the following releases. Our current packaging infrastructure provider is [Package Cloud](https://packagecloud.io).
```
Comment (joelhans) [inline packaging/installer/README.md]: Can you make this a new paragraph?

File: packaging/installer/README.md [hunk 11]
```diff
@@ -293,6 +293,28 @@ To apply the changes you made, you have to restart Netdata.
...
+Netdata is committed to support installation of our solution to all operating systems.
```
Comment (joelhans) [inline packaging/installer/README.md]: Start a new paragraph here, and combine the next two lines into this paragraph.

File: packaging/installer/README.md [hunk 12]
```diff
@@ -293,6 +293,28 @@ To apply the changes you made, you have to restart Netdata.
...
+We provide two separate repositories, one for our stable releases and one for our nightly releases.
+1. Stable releases
+
+   Our stable production releases are hosted in [netdata/netdata](https://packagecloud.io/netdata/netdata) repository of package cloud
+
+2. Nightly releases
+
+   Our latest releases are hosted in [netdata/netdata-edge](https://packagecloud.io/netdata/netdata-edge) repository of package cloud
+
+Visit the repository pages and follow the instructions for the quick repository set up
```
Comment (joelhans) [inline packaging/installer/README.md]: Try: `Visit the repository pages and follow the quick set-up instructions to get started.`

File: packaging/installer/README.md [hunk 13]
```diff
@@ -293,6 +293,28 @@ To apply the changes you made, you have to restart Netdata.
...
+Netdata is committed to support installation of our solution to all operating systems.
+This is a constant battle for Netdata, as we strive to automate and make things easier for our users.
+For the operating system support matrix, please visit our [distributions](../DISTRIBUTIONS.md) support page.
+
+We provide two separate repositories, one for our stable releases and one for our nightly releases.
+1. Stable releases
+
+   Our stable production releases are hosted in [netdata/netdata](https://packagecloud.io/netdata/netdata) repository of package cloud
+
+2. Nightly releases
```
Comment (joelhans) [inline packaging/installer/README.md]: For both these, just combine the lines together, otherwise the formatting is hard to follow when built.

```
1. Stable releases: Our stable production releases are hosted in [netdata/netdata](https://packagecloud.io/netdata/netdata) repository of Package Cloud.

2. Nightly releases: Our latest releases are hosted in [netdata/netdata-edge](https://packagecloud.io/netdata/netdata-edge) repository of Package Cloud.
```

File: packaging/installer/README.md (hunk 11 above)
Comment (joelhans) [reply packaging/installer/README.md]: Sorry, the code blocks aren't really working right. Start a new paragraph at `We have currently released...`

File: packaging/installer/README.md (hunk 10 above)
Comment (joelhans) [reply packaging/installer/README.md]: Meaning start a new paragraph at `We have currently released...`
